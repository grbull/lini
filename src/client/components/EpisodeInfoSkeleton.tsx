import cn from 'classnames';
import React, { ReactElement } from 'react';

import { InfoCard } from './InfoCard';
import { Skeleton } from './Skeleton';

export function EpisodeInfoSkeleton(): ReactElement {
  return (
    <InfoCard>
      <Skeleton className={cn('w-full', 'h-12')} />
      <Skeleton
        className={cn('w-full', 'h-0', 'relative', 'my-3')}
        style={{ paddingTop: '56.25%' }}
      />
      <Skeleton className={cn('w-full', 'h-16')} />
    </InfoCard>
  );
}
