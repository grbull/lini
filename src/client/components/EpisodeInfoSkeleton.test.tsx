/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { EpisodeInfoSkeleton } from './EpisodeInfoSkeleton';

describe('EpisodeInfoSkeleton Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<EpisodeInfoSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
