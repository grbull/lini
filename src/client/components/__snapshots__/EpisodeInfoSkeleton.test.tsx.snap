// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeInfoSkeleton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <div
      class="animate-pulse bg-gray-700 w-full h-12"
    />
    <div
      class="animate-pulse bg-gray-700 w-full h-0 relative my-3"
      style="padding-top: 56.25%;"
    />
    <div
      class="animate-pulse bg-gray-700 w-full h-16"
    />
  </div>
</DocumentFragment>
`;
